src/d3a/models/state.py

```diff
@@ -51,13 +51,13 @@ def get_energy_production_forecast_kWh(self, time_slot, default_value=None):
         assert self._energy_production_forecast_kWh[time_slot] >= -FLOATING_POINT_TOLERANCE
         if default_value is not None:
             return self._energy_production_forecast_kWh.get(time_slot, default_value)
-        return self._energy_production_forecast_kWh[time_slot]
+        return self._energy_production_forecast_kWh.get(time_slot, 0.)
```

Comment:
didnt you check this by moving https://github.com/gridsingularity/d3a/pull/1021/files#diff-00472b9024eb273db74d246aa396bcbb49a369f86334c3413c71a505a4994036R343 (this is what i got from the slack discussion)

Reply:
Hmm, will do it too but its not important at this point as at L51 at this point of commit would have already raised an exception

Reply:
ah, yes so you dont need this change then. I am being relentless here to show the reader of the code, that a `time_slot` should be indeed in `self._energy_production_forecast_kWh` at this point. RIght?